### PR TITLE
[fix] manual_seed{_all}: mem leak

### LINKED ISSUE #55768: torch.manual_seed leaks memory
## 🐛 Bug

Calling `torch.manual_seed(...)` (or `torch.cuda.manual_seed_all(...)`) leaks memory, quite a bit in fact. I am resetting the seed in my data loader for each sample and over the course of a few hours there are tens of gigabytes that are being leaked.

## To Reproduce

```
for i in range(1000000000): torch.manual_seed(i)
```

And watch the memory consumption of the process slowly creep up.

## Expected behavior

The memory consumption to remain constant, as is the case with `for i in range(1000000000): random.seed(i)`.

## Environment

```
PyTorch version: 1.8.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.19.6

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1650
Nvidia driver version: 460.39
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @pbelevich

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes: #55768

### CODE REVIEW

**Comment by kshitij12345** [comment]
Not sure about testing this. 

I did see that there are tests for the API itself (but nothing related to the laziness)

Thanks!

Ref:
https://github.com/pytorch/pytorch/blob/773a8eede437de972311b21824694e4fb120f6a2/test/test_cuda.py#L653

https://github.com/pytorch/pytorch/blob/773a8eede437de972311b21824694e4fb120f6a2/test/test_torch.py#L773

**Comment by ezyang** [comment]
In this case a manual test for the original issue I think would be easiest

**Review by ezyang — APPROVED** [review]

**Comment by kshitij12345** [comment]
<details>

<summary> Local Test </summary>

Code
```python
import torch
import os
import psutil

print("Process Id:", os.getpid())

def memory_usage_psutil():
    # return the memory usage in percentage like top
    process = psutil.Process(os.getpid())
    mem = process.memory_percent()
    return mem

for i in range(1000000000):
    if (i % 1e5 == 0):
        print(memory_usage_psutil(), "%")
    torch.manual_seed(i)
```

Before:
```
Process Id: 1589184
0.10165690276095886 %
0.1512824260005906 %
0.20143247745286674 %
```

After:
```
Process Id: 1591162
0.1019252307772538 %
0.1019252307772538 %
0.1019252307772538 %
```

</details>
